evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -1024,3 +1076,129 @@ def test_partial_dependence_respect_grid_resolution(fraud_100):
     dep = partial_dependence(pl, X, features="provider", grid_resolution=20)
     assert dep.shape[0] == X["provider"].nunique()
     assert dep.shape[0] != max(X.ww.select("categorical").describe().loc["unique"]) + 1
+
+
+@pytest.mark.parametrize("problem_type", [ProblemTypes.BINARY, ProblemTypes.MULTICLASS])
+def test_graph_partial_dependence_ice_plot(problem_type, test_pipeline, logistic_regression_multiclass_pipeline_class):
+    go = pytest.importorskip(
+        "plotly.graph_objects",
+        reason="Skipping plotting test because plotly not installed",
+    )
+
+    if problem_type == ProblemTypes.MULTICLASS:
+        test_pipeline = logistic_regression_multiclass_pipeline_class(
+            parameters={"Logistic Regression Classifier": {"n_jobs": 1}}
+        )
+        X, y = load_wine()
+        feature = "ash"
+    else:
+        X, y = load_breast_cancer()
+        feature = "mean radius"
+    clf = test_pipeline
+    clf.fit(X, y)
+
+    fig = graph_partial_dependence(
+        clf, X, features=feature, grid_resolution=20, kind="both"
+    )
+    assert isinstance(fig, go.Figure)
+    fig_dict = fig.to_dict()
+    assert (
+        fig_dict["layout"]["title"]["text"]
+        == f"Partial Dependence of '{feature}' <br><sub>Including Individual Conditional Expectation Plot</sub>"
+    )
+    n_classes = (len(y.unique()) if problem_type == ProblemTypes.MULTICLASS else 1)
+    assert len(fig_dict["data"]) == len(X) * n_classes + n_classes
+    expected_label = "Individual Conditional Expectation" if problem_type == ProblemTypes.BINARY else "Individual Conditional Expectation: class_0"
+    assert fig_dict["data"][0]["name"] == expected_label
+    assert fig_dict["data"][-1]["name"] == "Partial Dependence" if problem_type == ProblemTypes.BINARY else "Partial Dependence: class_2"
+
+    avg_dep_data, ind_dep_data = partial_dependence(
+        clf, X, features=feature, grid_resolution=20, kind="both"
+    )
+    assert np.array_equal(fig_dict["data"][-1]["x"], avg_dep_data["feature_values"][:len(fig_dict["data"][-1]["x"])].values)
+
+    if problem_type == ProblemTypes.BINARY:
+        assert np.array_equal(
+            fig_dict["data"][-1]["y"], avg_dep_data["partial_dependence"].values
+        )
+    else:
+        class_2_data = avg_dep_data[avg_dep_data['class_label'] == 'class_2']["partial_dependence"].values
+        assert np.array_equal(
+            fig_dict["data"][-1]["y"], class_2_data
+        )
+
+    for i in range(len(X)):
+        assert np.array_equal(fig_dict["data"][i]["x"], ind_dep_data["feature_values"])
+        if problem_type == ProblemTypes.MULTICLASS:
+            window_length = len(ind_dep_data[f"Sample {i}"].values)
+            data = np.concatenate([fig_dict["data"][i]["y"][:window_length],
+                                   fig_dict["data"][i + len(X) + 1]["y"][:window_length],
+                                   fig_dict["data"][i + 2 * len(X) + 2]["y"][:window_length]])
```

Comment:
For this particular test, I ended up doing some weird matrix math to wrangle the plotly data. For each sample, this snippet essentially combines the ICE data for each class to one array and compares it to the data for that particular sample in the `partial_dependence()` output data.  The +1 and +2 accounts for the additional PD trace that is in each plot.